Source/WebCore/PlatformWinCairo.cmake

```diff
@@ -78,6 +80,12 @@ target_link_options(MediaFoundation INTERFACE
     /DELAYLOAD:mfplat.dll
 )
 
+if (WebCore_LIBRARY_TYPE STREQUAL SHARED)
+    list(APPEND WebCore_PRIVATE_LIBRARIES MediaFoundation)
+else ()
+    list(APPEND WebCore_LIBRARIES MediaFoundation)
```

Comment:
Why should `MediaFoundation` be `PUBLIC` link type unless SHARED?

Reply:
https://bugs.webkit.org/show_bug.cgi?id=214170 is the problem why MediaFoundation shouldn't link to WebCore. Does this patch regress?

Reply:
I double checked with a build and `WebCore_PRIVATE_LIBRARIES` is correct and there is no linker error.

Reply:
weird. I think we should revert the change of https://bugs.webkit.org/show_bug.cgi?id=214170 .

Reply:
Let's remove these lines. https://github.com/WebKit/WebKit/blob/1713bed5d0c732d56f06c5947117844dbdc4f765/Source/WebKitLegacy/PlatformWin.cmake#L12-L15

Reply:
Done and https://ews-build.webkit.org/#/builders/12/builds/101216 has no problems

Reply:
https://github.com/WebKit/WebKit/blob/cb60df93ab8cb5be5252409d0f934a0b5fabe082/Source/WebCore/PlatformWinCairo.cmake#L61-L65
If we don't see the linker warning, we should remove this comment. And, we can remove the MediaFoundation INTERFACE library.